cylc/uiserver/scripts/hub.py

```diff
@@ -38,6 +80,6 @@ def main(*args):
     # set an env var flag to help load the config
     os.environ['CYLC_HUB_VERSION'] = __version__
     try:
-        hub_main(args)
+        JupyterHub.launch_instance(args)
```

Comment:
Another, perhaps smaller workaround, is to act as if `launch_instance` only supports sys.argv (since that's technically true due to https://github.com/jupyterhub/jupyterhub/pull/4039):

```python
import sys
from unittest.mock import patch
# set sys.argv for launch_instance to parse
with patch.object(sys, "argv", [sys.argv[0]] + list(args)):
    JupyterHub.launch_instance()
```

Reply:
Thanks for the advice @minrk - considering that approach now!

Reply:
I've checked this works, although it took a bit of investigation see how `sys.argv` gets in there: JupyterHub is a `traitlets.config Application`, and during initialization the `Application.parse_command_line()` method does it.

Reply:
Yes, most command-line entrypoints (including argparse itself) look to `sys.argv` if argv aren't explicitly given (the default). Explicit argv is the minority case, where CLI args are passed but not on the actual CLI.